### PR TITLE
Complete code-level documentation and examples

### LINKED ISSUE #17: Improve documentation and usage examples
Following the example of `vm-memory`, `linux-loader` should have extensively documented public-facing APIs, with unit tested examples.

### PR DESCRIPTION
**Pull request by aghecenco**
Fixes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,58 +1,128 @@
...
-Our Continuous Integration (CI) pipeline is implemented on top of
-[Buildkite](https://buildkite.com/).
-For the complete list of tests, check our
-[CI pipeline](https://buildkite.com/rust-vmm/rust-vmm-ci).
+### Loading the kernel command line
 
-Each individual test runs in a container. To reproduce a test locally, you can
-use the dev-container on both x86 and arm64.
+After all the devices have been configured, no further changes can come to the
+kernel command line. It can therefore be loaded in guest memory.
```
[the remark this answers is not in the document]
Comment (aghecenco) [reply README.md:81]: I'll rephrase to suggest that device configuration and cmdline building happens here (between memory setup + kernel load and reg/bootparams setup). Too much details about MMIO and PCI might make the guide harder to follow as they're unrelated to `linux-loader` anyway,

File: docs/TESTING.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+           --security-opt seccomp=unconfined \
+           --volume $(pwd):/linux-loader \
+           rustvmm/dev:v${container_version}
+cd linux-loader/
+cargo test
+```
+
+### bzImage test
+
+As we don't want to distribute an entire kernel bzImage, the `load_bzImage`
```
[the remark this answers is not in the document]
Comment (aghecenco) [reply docs/TESTING.md:25]: Think you caught that in https://github.com/rust-vmm/linux-loader/issues/34

[the remark this answers is not in the document]
Comment (andreeaflorescu) [reply docs/TESTING.md:25]: Not on purpose :))

**Comment by aghecenco** [comment]
@andreeaflorescu feedback addressed (I hope), PTAL

**Pull request merged**